### ISSUE TITLE
optional embed title field for ctx.maybe_send_embed()

### ISSUE THREAD

**Issue by Obi-Wan3** [Cog-Creators/Red-DiscordBot#4879]
### What component of Red (cog, command, API) would you like to see improvements on?

ctx.maybe_send_embed()

### Describe the enhancement you're suggesting.

In `ctx.maybe_send_embed()`, accept an optional kwarg for the title of the embed if it is able to be sent. This function has proved to be very helpful and takes the place of a manually created check-for-embed-permissions-and-send function in every cog, and it would be great to have this additional feature for when a title would be nice to have but is not required. Additionally, this could be expanded to accept every field a typical discord.Embed would accept, but currently the title field would be most useful.

**Comment by Flame442** [comment]
What would the function do with the title field if the message could NOT be sent as an embed? Ignore it, prepend it, etc?

**Comment by kingslayer268** [comment]
prepending it would probably look ugly, best would be to ignore it

**Comment by Obi-Wan3** [comment]
I agree with kingslayer268, I imagined this for scenarios where it's a **nice** to have but not **need** to have.

**Comment by yamikaitou** [comment]
+1 to being able to pass an Embed object for use (instead of individual fields).

**Comment by Obi-Wan3** [comment]
Yami, if that were the case would the fallback be just sending the description? If so, I think that's fine as well 👍

**Comment by yamikaitou** [comment]
I'd probably rather the fallback be the existing message arg. That way people can customize it as they like for complex embeds

**Comment by Obi-Wan3** [comment]
I think it would be good to still be able to use either the message arg as the embed desc (as right now) **or** the embed desc as the fallback message, as part of the reason this function is so helpful for me right now is that I only have to pass in 1 thing instead of both, otherwise it would only be 1 more step to add in a try-except myself instead.

**Comment by kingslayer268** [comment]
> +1 to being able to pass an Embed object for use (instead of individual fields).

I think there should be a completely different function for that, `ctx.maybe_send_embed()` is meant for simple messages

**Comment by TrustyJAID** [comment]
I like the idea of being similar to regular `Messageable.send` but instead of passing the content arg as the messages plain-text response it chooses to use either the plain-text or the embed field depending on embed-link permissions. Keeping backwards compatibility is a must though but adding an optional embed argument with the function sounds like it would work well.

**Comment by yamikaitou** [comment]
So my thought/thinking is this

Add an optional (kw)arg to the existing function that accepts a discord.Embed object (similar to `Messageable.send` as Trusty said).
If the bot can send an Embed, it will use the new arg if provided, otherwise it will embed the existing `message` arg as it currently does
If the bot can not send an Embed, it will use the existing message arg as normal

**Comment by Obi-Wan3** [comment]
I was thinking similar to what Yami just said, but the message arg is made optional as well so in cases where that is *not* provided and an embed is, then the embed's description would be used (of course, one of the two would need to be provided).

**Comment by Drapersniper** [comment]
I thought this method was supposed to be embed independent, if we were to pass a title, how would the title and content be shown when embed are disabled ? Same goes any other embed part.

If we want to just send based on context permissions perhaps we should just add a public accessible attribute to context to flag if embeds are allowed in the context and let the develop handle it how they want (to avoid building both an embed object and a text object) the behaviour in this method should be consistent across the board (the final message)

**Comment by TrustyJAID** [comment]
I don't think pulling the embed description is worth it since the user can provide that themselves with the proposed alternative by doing `await ctx.maybe_send_embed(embed.description, embed=embed)`. The original arg's should still exist as they do now just add the new optional (kw)arg for a full embed to use.

**Comment by Obi-Wan3** [comment]
Thanks Trusty for explaining that, I agree now that the original message arg can stay as is.
